### PR TITLE
6 info about unreleased changes

### LINKED ISSUE #6: Projects tab: information about unreleased changes
Each project in the repos.json should contain information about unreleased changes, i.e. pushes to `main` that are not part of a release. This info should also contain the number of unreleased commits.

### PR DESCRIPTION
**Pull request by mweidling**
This PR introduces a new mandatory attribute, `unreleased_changes`, to the `Repo` class. It denoted how many commits have been made since the last release/tag or how many commits a repository has if no tags are given.

Closes #6.

This PR has to be handled after #7.

### CODE REVIEW

**Review by kba — APPROVED** [review]

**Comment by kba** [comment]
Did you rebase between #7 and #10/#12? Because GH still shows all the commits in both later PR.

**Comment by mweidling** [comment]
> Did you rebase between #7 and #10/#12? Because GH still shows all the commits in both later PR.

Mmh, I usually don't rebase but merge, see [9837d71](https://github.com/OCR-D/kwalitee-dashboard-back-end/pull/12/commits/9837d7122681e15ba08a63bcccd1e1bcae206820). Is it better practice to rebase?

File: kwalitee/repo.py [hunk 1]
```diff
@@ -78,6 +80,36 @@ def get_project_type(self):
...
+        complete_refs = self._run('git show-ref --tag')
+        formatted = complete_refs.stdout.strip()
+        item_list = list(formatted.split('\n'))
+        latest_release = item_list[-1]
+        pattern = 'v?[0-9]+\.[0-9]+.*?$'
+        match = re.findall(pattern, latest_release)
+
+        if match:
+            return match[0]
+        else:
```
[the remark this answers is not in the document]
Comment (mweidling) [reply kwalitee/repo.py:116]: See [10cee35](https://github.com/OCR-D/kwalitee-dashboard-back-end/pull/12/commits/10cee352abb1c3033bbfd918b691f8c1865a2ea1).

File: kwalitee/repo.py [hunk 2]
```diff
@@ -78,6 +80,36 @@ def get_project_type(self):
...
+                dist = self._run(cmd).stdout.strip()
+                return int(dist)
+            else:
+                total_no_of_commits = self._run('git rev-list --count HEAD').stdout.strip()
+                return int(total_no_of_commits)
+
+    def _get_latest_tag(self):
+        complete_refs = self._run('git show-ref --tag')
+        formatted = complete_refs.stdout.strip()
+        item_list = list(formatted.split('\n'))
```
[the remark this answers is not in the document]
Comment (mweidling) [reply kwalitee/repo.py]: See [10cee35](https://github.com/OCR-D/kwalitee-dashboard-back-end/pull/12/commits/10cee352abb1c3033bbfd918b691f8c1865a2ea1).

**Pull request merged**